### ISSUE TITLE
Syncthing does not use correct folder

### ISSUE THREAD

**Issue by VBMCBoy** [syncthing/syncthing#6298]
When receiving a folder from another device, choosing a folder does not actually select that folder on disk. E.g. when receiving a folder called "example", and choosing "\~/Documents/example" as location, the directory is actually synced to "\~/example" instead.

corresponding log output:
```
2020-01-28T19:00:01.3057264+01:00: 
Unexpected folder "Camera" (lg-h815_kw59-photos) sent from device "...-MU6YUQ6"; ensure that the folder exists and that this device is selected under "Share With" in the folder configuration.

2020-01-28T19:01:05.582940525+01:00: 
Adding folder "Camera" (lg-h815_kw59-photos)

2020-01-28T19:01:05.58302048+01:00: 
No stored folder metadata for "lg-h815_kw59-photos": recalculating

2020-01-28T19:01:05.583479237+01:00: 
Connection to ...-MU6YUQ6 at [fe80::22d9:aae5:b71f:92ba%wlp5s0]:22000-[fe80::7af8:82ff:fece:1908%wlp5s0]:37405/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256 closed: reading length: read tcp [fe80::22d9:aae5:b71f:92ba%wlp5s0]:22000->[fe80::7af8:82ff:fece:1908%wlp5s0]:37405: use of closed network connection

2020-01-28T19:01:05.590758817+01:00: 
Ready to synchronize "Camera" (lg-h815_kw59-photos) (sendreceive)

2020-01-28T19:01:13.257561575+01:00: 
Completed initial scan of sendreceive folder "Camera" (lg-h815_kw59-photos)

2020-01-28T19:02:02.510558394+01:00: 
Established secure connection to ...-MU6YUQ6 at [fe80::22d9:aae5:b71f:92ba%wlp5s0]:22000-[fe80::7af8:82ff:fece:1908%wlp5s0]:37414/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256

2020-01-28T19:02:02.510716539+01:00: 
Device ...-MU6YUQ6 client is "syncthing v1.3.3" named "LG G4" at [fe80::22d9:aae5:b71f:92ba%wlp5s0]:22000-[fe80::7af8:82ff:fece:1908%wlp5s0]:37414/tcp-server/TLS1.3-TLS_AES_128_GCM_SHA256

2020-01-28T19:02:02.518604676+01:00: 
Device ...-MU6YUQ6 folder "Camera" (lg-h815_kw59-photos) has mismatching index ID for us (0xC79690335B78EF05 != 0xF5320D1F85FDC121)

2020-01-28T19:02:02.518630333+01:00: 
Device ...-MU6YUQ6 folder "Camera" (lg-h815_kw59-photos) has a new index ID (0x5A36F71183EB9ADC)
```

Screenshots:
Adding a folder called Camera to `/run/media/kay/..../Backup`...
![image](https://user-images.githubusercontent.com/15034727/73291582-bb6ae700-4200-11ea-982e-10ed00de8f30.png)
Instead adds it to `/home/kay/Camera`
![image](https://user-images.githubusercontent.com/15034727/73291622-d3db0180-4200-11ea-8833-52331ab91ce9.png)

#### receiving end
Syncthing Version: v1.3.3
OS Version: Arch Linux
#### sending end
Syncthing Version: v1.3.3
OS Version: Android 9
(however the same bug happens when sending from another Arch Linux computer with v1.3.3

**Comment by AudriusButkevicius** [comment]
I don't know where that select folder button comes from but that's not a button syncthing provides, that's most likely the cause.

**Comment by calmh** [comment]
Yeah this does not look like stock Syncthing. Can you reproduce using the regular Syncthing UI? If so, yell to reopen.

**Comment by VBMCBoy** [comment]
I just tried it in Firefox and there the button does not appear (and typing in the path works), so apparently when viewing the UI in [syncthingtray](https://github.com/Martchus/syncthingtray) (using its internal browser) it adds that button. So I guess that is just a problem with syncthingtray. I didn't know that it would do that, otherwise I would've added that information to the issue.
I'm very sorry if I caused any confusion, I should've tried it in a normal browser too. Thank you very much for your time.
